dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/relationship/RelationshipsExportControllerTest.java

```diff
@@ -588,37 +588,41 @@ void getRelationshipsByTrackedEntityRelationshipsNoAccessToRelationshipItemTo()
   }
 
   @Test
-  void getRelationshipsByTrackedEntityRelationshipsNoAccessToBothRelationshipItems() {
+  void shouldReturnForbiddenWhenGetRelationshipsWithBothEndsNotAccessible() {
```

Comment:
Now that you've added the logic + unit tests, do we need the `shouldReturnForbiddenWhenGetRelationshipsWithBothEndsNotAccessible` and `shouldReturnForbiddenWhenGetRelationshipsByNotAccessibleTrackedEntity`?

I am worried we have too many overlapping tests => many failing for the same reason 😅

Reply:
I agree, we can remove some overlapping tests.
I will remove `shouldReturnForbiddenWhenGetRelationshipsWithBothEndsNotAccessible`.
I still think that `shouldReturnForbiddenWhenGetRelationshipsByNotAccessibleTrackedEntity` make sense to be there to show the different behavior from `getRelationshipsByTrackedEntityRelationshipsNoAccessToRelationshipItemTo`.

Reply:
is `shouldReturnForbiddenWhenGetRelationshipsByNotAccessibleTrackedEntity` then specific to the `from` if so its name should probably be symmetrical to `getRelationshipsByTrackedEntityRelationshipsNoAccessToRelationshipItemTo`